commands/project_eval.go

```diff
@@ -0,0 +1,86 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/phase2/rig/util"
+)
+
+// ProjectEval wrapps the evaluation of project scripts.
+// It mimics command struct except with unexported values.
+type ProjectEval struct {
+	out    *util.RigLogger
+	config *ProjectConfig
+}
+
+// ProjectScriptRun takes a ProjectScript configuration and executes it per
+// the definition of the project script and bonus arguments from the extra parameter.
+// Commands are run from the directory context of the project if available.
+// Use ProjectScriptRun to run a comman for potential user interaction.
+func (p *ProjectEval) ProjectScriptRun(script *ProjectScript, extra []string) int {
+	p.out.Verbose("Initializing project script '%s': %s", script.ID, script.Description)
+	p.addCommandPath()
+	dir := p.GetWorkingDirectory()
+	shellCmd := p.GetCommand(script.Run, extra, dir)
+	p.out.Verbose("Evaluating Script '%s'", script.ID)
+	return util.PassthruCommand(shellCmd)
+}
+
+// ProjectScriptResult matches ProjectScriptRun, but returns the data from the
+// command execution instead of "streaming" the result to the terminal.
+func (p *ProjectEval) ProjectScriptResult(script *ProjectScript, extra []string) (string, int, error) {
+	p.out.Verbose("Initializing project script '%s': %s", script.ID, script.Description)
+	p.addCommandPath()
+	dir := p.GetWorkingDirectory()
+	shellCmd := p.GetCommand(script.Run, extra, dir)
+	p.out.Verbose("Evaluating Script '%s'", script.ID)
+	return util.CaptureCommand(shellCmd)
```

Comment:
Since this is exactly the same as ProjectSriptRun, what is we passed in the output writer and if it is null Passthru, otherwise capture? (not sure it makes sense, but a thought)

Reply:
The thing is I want to capture stdout, stderr, and the exit code. The timing of when or if we output the result of the command depends on the caller. For an example of when this matters, see the project doctor PR which outputs the stdout verbosely, and may not even do that in the future.